src/test/java/com/microsoft/azure/relay/SendReceiveTest.java

```diff
@@ -326,18 +392,28 @@ private static void sendResponseMessages(RelayedHttpListenerContext context, Str
 		}
 	}
 	
-	private static String readStringFromStream(InputStream inputStream) throws IOException {
+	private static byte[] readBytesFromStream(InputStream inputStream) throws IOException {
 		if (inputStream == null) {
-			return emptyStr;
+			return new byte[0];
 		}
 		
-		StringBuilder builder = new StringBuilder();
-		String inputLine;
-		try (BufferedReader inStream = new BufferedReader(new InputStreamReader(inputStream))){
-			while ((inputLine = inStream.readLine()) != null) {
-				builder.append(inputLine);
+		ByteArrayOutputStream byteStream = new ByteArrayOutputStream();
+		while (true) {
+			int i = inputStream.read();
+			if (i == -1) {
+				break;
+			} else {
+				byteStream.write(i);
 			}
 		}
-		return builder.toString();
+		return byteStream.toByteArray();
+	}
+	
+	private byte[] getByteSet() {
```

Comment:
>getByteSet [](start = 16, length = 10)

Do all these byte-level tests use exactly 256bytes?

Reply:
All 256 bytes are checked for equality

---
In reply to: [294017638](https://github.com/Azure/azure-relay-java/pull/47#discussion_r294017638) [](ancestors = 294017638)

Reply:
At some point it would be nice to check that an odd number of bytes work. 1, 255, 1024, 4k, 64k-1, 64k, 64k+1, etc. In the future consider having getByteSet take a size and fill an array of that size with this repeating byte pattern.

---
In reply to: [294017638](https://github.com/Azure/azure-relay-java/pull/47#discussion_r294017638) [](ancestors = 294017638)

Reply:
Do you mean replace the current tests using small/large strings with these bytes repeated? I'm not sure what kind of difference it would make with different sizes like 1024 vs 4096

---
In reply to: [294377702](https://github.com/Azure/azure-relay-java/pull/47#discussion_r294377702) [](ancestors = 294377702,294017638)

Reply:
First of all I'm trying to get you outside of testing strings. The library deals with bytes not Strings as near AFAICT. Strings convert to bytes in a MANY different ways (which are outside the scope of the library here).

I want to ensure if the library is given these 11 bytes:
0xFF 0xFE 0x65 0x00 0x66 0x00 0x67 0x00 0x00 0x00 0x00 (UTF-16LE with byte order mark string of "ABC" plus extra 0x00 0x00 terminator plus one extra 0x00 byte to make it an odd number)

that our library doesn't convert it to something else like these 4 bytes which once converted to String are equivalent to the original 10 bytes:
0x65 0x66 0x67 0x00            (ASCII or UTF-8 with no byte order mark, lost trailing 0x00 bytes).

---
In reply to: [295527790](https://github.com/Azure/azure-relay-java/pull/47#discussion_r295527790) [](ancestors = 295527790,294377702,294017638)